### PR TITLE
Release nydus-snapshotter packages with different architectures

### PR DESCRIPTION
**Pull request by sctb512**
Related to https://github.com/containerd/nydus-snapshotter/issues/548

Expected packages: https://github.com/sctb512/nydus-snapshotter/releases/tag/v0.9.2

### CODE REVIEW

**Comment by fidencio** [comment]
@sctb512, I'm closing #548 in favour of this one, as this one is way more complete. :-)

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -13,6 +13,21 @@ env:
...
+          - build-os: linux
+            build-arch: arm64
+          - build-os: linux
+            build-arch: s390x
+          - build-os: linux
+            build-arch: ppc64le
+          - build-os: linux
+            build-arch: riscv64
+          - build-os: linux
+            build-arch: static
```
Comment (fidencio) [inline .github/workflows/release.yml:30]: Do we actually need this?
I would assume that we don't, as for all the arches we're actually statically linking the binaries.

Another thing to be considered, is whether we need changes in the makefiles for building the content for different arches, as the changes done on #552.

**Comment by fidencio** [comment]
@stevenhorsman, as Choi is on vacation, would you mind to give it a try on the binaries released for s390x and make sure those are working as expected?

**Comment by BbolroC** [comment]
Hi, I've tested the binaries. All looks good, but `optimizer-server` as follows:

```
$ file optimizer-server
optimizer-server: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=21ade306589fc7f613b58e8f5bcfea0b6f8c7150, for GNU/Linux 3.2.0, with debug_info, not stripped
```

**Comment by sctb512** [comment]
> Hi, I've tested the binaries. All looks good, but `optimizer-server` as follows:
> 
> ```
> $ file optimizer-server
> optimizer-server: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=21ade306589fc7f613b58e8f5bcfea0b6f8c7150, for GNU/Linux 3.2.0, with debug_info, not stripped
> ```

Thanks for testing. Need to add `EXTRA_RUSTFLAGS` to build rust the `optimizer-server`. I will update it soon.

**Comment by sctb512** [comment]
Updated, `strip` is not supported for the result of cross compilation, so I skip it. PTAL @BbolroC https://github.com/sctb512/nydus-snapshotter/actions/runs/6734057338

**Comment by BbolroC** [comment]
Hi, the test results are as follows:

```
~/bin$ ls
containerd-nydus-grpc  nydus-overlayfs  optimizer-nri-plugin  optimizer-server
~/bin$ ls -1 | xargs -i{} file {}
containerd-nydus-grpc: ELF 64-bit MSB executable, IBM S/390, version 1 (SYSV), statically linked, Go BuildID=D9FMM2zeoLZRudY9jmk3/w2Ubn4ekE7PDXkW8KUwC/hj2l8NJJPWnLMuVNvUO6/CCeUYXb37ebI3TvRpL7k, stripped
nydus-overlayfs: ELF 64-bit MSB executable, IBM S/390, version 1 (SYSV), statically linked, Go BuildID=NJHuJic9uc6alIavAiRK/KiGolH9qCL_OQVxPxzfi/KM9yYXMKgWR3MSc5Nr-i/o7iFDb_RRuAgtRVaF_2a, stripped
optimizer-nri-plugin: ELF 64-bit MSB executable, IBM S/390, version 1 (SYSV), statically linked, Go BuildID=pLWwkeK1wDheG0Lr94z7/E0s98LRkYKZ00rvJ3G3z/WdRYAJ_wHeV0vejegCFS/hNz8MRM1B7gDz0xsL3TZ, stripped
optimizer-server: ELF 64-bit MSB shared object, IBM S/390, version 1 (SYSV), dynamically linked, interpreter /lib/ld64.so.1, BuildID[sha1]=42f7921d6175f7c1c5c881690126d85301feeac2, for GNU/Linux 3.2.0, with debug_info, not stripped
~/bin$ ./optimizer-server --help
./optimizer-server: /lib/s390x-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by ./optimizer-server)
./optimizer-server: /lib/s390x-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by ./optimizer-server)
./optimizer-server: /lib/s390x-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by ./optimizer-server)
```

Only the binary `optimizer-server` is built as `dynamically linked` and `not stripped`. Is this what you expected? Thanks.

**Comment by sctb512** [comment]
> Hi, the test results are as follows:
> 
> ```
> ~/bin$ ls
> containerd-nydus-grpc  nydus-overlayfs  optimizer-nri-plugin  optimizer-server
> ~/bin$ ls -1 | xargs -i{} file {}
> containerd-nydus-grpc: ELF 64-bit MSB executable, IBM S/390, version 1 (SYSV), statically linked, Go BuildID=D9FMM2zeoLZRudY9jmk3/w2Ubn4ekE7PDXkW8KUwC/hj2l8NJJPWnLMuVNvUO6/CCeUYXb37ebI3TvRpL7k, stripped
> nydus-overlayfs: ELF 64-bit MSB executable, IBM S/390, version 1 (SYSV), statically linked, Go BuildID=NJHuJic9uc6alIavAiRK/KiGolH9qCL_OQVxPxzfi/KM9yYXMKgWR3MSc5Nr-i/o7iFDb_RRuAgtRVaF_2a, stripped
> optimizer-nri-plugin: ELF 64-bit MSB executable, IBM S/390, version 1 (SYSV), statically linked, Go BuildID=pLWwkeK1wDheG0Lr94z7/E0s98LRkYKZ00rvJ3G3z/WdRYAJ_wHeV0vejegCFS/hNz8MRM1B7gDz0xsL3TZ, stripped
> optimizer-server: ELF 64-bit MSB shared object, IBM S/390, version 1 (SYSV), dynamically linked, interpreter /lib/ld64.so.1, BuildID[sha1]=42f7921d6175f7c1c5c881690126d85301feeac2, for GNU/Linux 3.2.0, with debug_info, not stripped
> ~/bin$ ./optimizer-server --help
> ./optimizer-server: /lib/s390x-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by ./optimizer-server)
> ./optimizer-server: /lib/s390x-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by ./optimizer-server)
> ./optimizer-server: /lib/s390x-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by ./optimizer-server)
> ```
> 
> Only the binary `optimizer-server` is built as `dynamically linked` and `not stripped`. Is this what you expected? Thanks.

Updated. It works as expected now.

**Comment by sctb512** [comment]
Need to remove the `static` package, as it is now the same as `x86-64`.

**Comment by BbolroC** [comment]
@sctb512 I've verified the binaries work on s390x. Thanks a lot! 😉

**Review by jiangliu — APPROVED** [review]

**Pull request merged**